lib/plugins/bed.js

```diff
@@ -147,15 +147,26 @@ function inject (bot) {
       }
 
       await bot.activateBlock(bedBlock)
+
+      const err = await waitUntilSleep()
+      if (err) {
+        throw new Error(err)
+      }
     }
   }
 
-  bot._client.on('game_state_change', (packet) => {
-    if (packet.reason === 0) {
-      // occurs when you can't spawn in your bed and your spawn point gets reset
-      bot.emit('spawnReset')
-    }
-  })
+  async function waitUntilSleep () {
+    return new Promise((resolve, reject) => {
+      const timeoutForSleep = setTimeout(() => {
+        resolve('bot is not sleeping')
```

Comment:
```suggestion
        reject('bot is not sleeping')
```
Shouldn't this be `reject`?

Reply:
I tried, but with reject don't received data,
Also they don't throw an error correctly

Reply:
With reject
![image](https://user-images.githubusercontent.com/20754836/167262384-4f18b755-8633-4e8e-93ce-aa41edc0f85b.png)

Reply:
try `reject(new Error('bot is not sleeping'))`

Reply:
@U5B you are great i learned a new thing :D